### PR TITLE
CLI and test things

### PR DESCRIPTION
**Pull request by yaleman**
Started poking around at some more test things for end-to-end testing, and had issues with a lack of response codes - added `handle_client_error` to the CLI which throws a non-zero exit code when we get a 500 back from the server.

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: scripts/test_run_release_server.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    echo "I'm not sure where you are, please run this from the root of the repository or the server/daemon directory"
+    exit 1
+fi
+
+mkdir -p /tmp/kanidm/
+
+echo "Generating certificates..."
+cargo run --bin kanidmd --release cert-generate --config ../../examples/insecure_server.toml
+echo "Running the server..."
+cargo run --bin kanidmd --release server --config ../../examples/insecure_server.toml &
```
Comment (Firstyear) [inline scripts/test_run_release_server.sh:31]: What's the error condition here? Shouldn't we try an admin pw recovery and use the success/failure of that to determine is this is a pass/fail?

Comment (yaleman) [reply scripts/test_run_release_server.sh:31]: It starts the server so the setup dev environment script can work so it tests "live" database and web things... which we've missed in testing before 😨

Comment (Firstyear) [reply scripts/test_run_release_server.sh:31]: Yeah, but my point here is that if the server fails to start do we detect or error here in this script?

Comment (yaleman) [reply scripts/test_run_release_server.sh:31]: Ah... I'll double check that.

Comment (yaleman) [reply scripts/test_run_release_server.sh:35]: Sorted... I think!

**Review by Firstyear — APPROVED** [review]

**Pull request merged**